syntax = "proto3";

package fixbug;

//定义下面的方法
option cc_generic_services = true;

message ResultCode{
    int32 errcode = 1;
    bytes errmsg = 2;
}


message GetFriendListRequest{
    uint32 userid = 1;
}


message User{
    bytes name = 1;
    uint32 age = 2;
    enum Sex{
        MAN = 0;
        WOMAN = 1;
    }
    Sex sex = 3;
}

message GetFriendListResponese{
    ResultCode result = 1;
    repeated User friend_list = 2;
}


service FriendServiceRpc{
    rpc GetFriendList(GetFriendListRequest) returns (GetFriendListResponese);
}
